### PR TITLE
Create an abstraction for calling the api

### PR DESCRIPTION
**Pull request by Leonils**
## Purpose

The goal of this PR is to introduce tools to make api calls from the components, in a way that can easily be overriden by the developers using the lib, by using a dependency injection patern

## Proposal

This PR introduce following components

- A provider, to wrap the app in
- 3 controllers : a mock for the storybook (console.log + fixtures), , a mock for tests (jest.fn), and a controller to call the api
- a hook to get the controller

And demonstrate how to use them in component TestButton and its tests

### CODE REVIEW

**Comment by Leonils** [comment]
Should fix issue #19

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,3 +11,8 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 - Add models to handle rooms, meetings, labels and groups
 
 ## [Unrealeased]
+
+### [Added]
+
+- An abstraction of the backend easily mocked and tested
+  that also can be overrided
```
Comment (sampaccoud) [inline CHANGELOG.md:18]:
```suggestion
  that also can be overriden
```

File: src/frontend/demo/package.json [hunk 2]
```diff
@@ -5,6 +5,7 @@
   "private": false,
   "dependencies": {
     "@jitsi-magnify/core": "0.1.0",
+    "axios": "^0.27.2",
```
Comment (sampaccoud) [inline src/frontend/demo/package.json:8]: We must pin dependencies to ensure reproducibility. We can add renovate to keep our dependencies upgraded with automatic Pull Requests:
```suggestion
    "axios": "0.27.2",
```

File: src/frontend/magnify/package.json [hunk 3]
```diff
@@ -32,6 +32,7 @@
     "@storybook/testing-library": "0.0.11",
     "@testing-library/jest-dom": "5.16.4",
     "@testing-library/react": "13.3.0",
+    "@testing-library/user-event": "^14.2.0",
```
Comment (sampaccoud) [inline src/frontend/magnify/package.json:35]:
```suggestion
    "@testing-library/user-event": "14.2.0",
```

File: src/frontend/magnify/package.json [hunk 4]
```diff
@@ -67,6 +68,7 @@
     "src/*"
   ],
   "peerDependencies": {
+    "axios": "^0.27.2",
```
Comment (sampaccoud) [inline src/frontend/magnify/package.json:71]:
```suggestion
    "axios": "0.27.2",
```

File: src/frontend/package.json [hunk 5]
```diff
@@ -25,7 +25,8 @@
     "format-demo:write": "cd demo && yarn format:write",
     "format-magnify:write": "cd magnify && yarn format:write",
     "format:check": "yarn format-magnify:check && yarn format-demo:check",
-    "format:write": "yarn format-magnify:write && yarn format-demo:write"
+    "format:write": "yarn format-magnify:write && yarn format-demo:write",
+    "verify-all": "yarn format:check & yarn lint & yarn build & yarn test"
```
Comment (sampaccoud) [inline src/frontend/package.json:29]: this could be a separate commit?

File: src/frontend/demo/src/index.tsx [hunk 6]
```diff
@@ -22,12 +27,17 @@ async function render() {
     translatedMessages = {};
   }
 
+  // Create the controller
+  const controller = new DefaultController({ url: 'http://localhost:3000' });
```
Comment (sampaccoud) [inline src/frontend/demo/src/index.tsx:31]: this url will have to be a variable?
```suggestion
  const controller = new DefaultController({ url: 'http://localhost:3000' });
```

File: src/frontend/magnify/src/controller/default.controller.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import Controller from './interface';
+import axios, { AxiosInstance } from 'axios';
+
+interface IDefaultControllerOptions {
```
Comment (sampaccoud) [inline src/frontend/magnify/src/controller/default.controller.ts:4]: Is there a mispell?
```suggestion
interface DefaultControllerOptions {
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -11,3 +11,8 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 - Add models to handle rooms, meetings, labels and groups
 
 ## [Unrealeased]
+
+### [Added]
```
Comment (sampaccoud) [inline CHANGELOG.md:15]: My understanding was that you wanted to use reactQuery. Is it still the plan?

File: src/frontend/magnify/src/controller/default.controller.ts (hunk 7 above)
Comment (Leonils) [reply src/frontend/magnify/src/controller/default.controller.ts:4]: Indeed. I'm used to name my interfaces beginning by a I, but we didn't follow this rule in this project, so I remove it.

File: src/frontend/demo/src/index.tsx (hunk 6 above)
Comment (Leonils) [reply src/frontend/demo/src/index.tsx:31]: Yes, it will

File: src/frontend/package.json (hunk 5 above)
Comment (Leonils) [reply src/frontend/package.json:29]: Yes

File: src/frontend/demo/src/index.tsx (hunk 6 above)
Comment (Leonils) [reply src/frontend/demo/src/index.tsx:31]: Just note that right now, the whole demo frontend is just a sandbox to demonstrate and try the lib, so I think until we connect the backend, it can wait.

File: CHANGELOG.md (hunk 8 above)
Comment (Leonils) [reply CHANGELOG.md:15]: reactQuery will be another layer to manage the result to api calls, when this is needed. It will be used later

**Review by lunika — APPROVED** [review]
